src/frontend/e2e/pages/upload.ts

```diff
@@ -0,0 +1,86 @@
+import { Page } from "@playwright/test";
+import { faker } from "@faker-js/faker";
+import { SampleData } from "../utils/schemas/sampleData";
+import { UploadData } from "../utils/schemas/uploadData";
+import * as path from 'path';
+
+
+export class UploadSample {
+  public static async uploadFiles(
+    page: Page,
+    uploadData: UploadData,
+  ): Promise<any> {
+    //click upload button
+    await page.locator("a[href$='/upload/step1']").click();
+    await UploadSample.selectSample(page,uploadData.dataFile);
+    // complete form
+    for(let i = 0; i < uploadData.samples.length; i++){
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='publicId']`).type(uploadData.samples[i].publicId);
```

Comment:
Pls put a one-liner to comment so we know which field is being filled/populated
Also put a TODO not to replace selectors with IDs